packages/tools/src/convertPathsIntoSubObjects.ts

```diff
@@ -0,0 +1,39 @@
+export function convertPathsIntoSubObjects(object: Record<string, any>): Record<string, any> {
+	const newObject: Record<string, any> = {};
+
+	for (const key in object) {
+		if (!(key in object)) {
+			continue;
+		}
+		const value = object[key];
```

Comment:
Is this continue reachable? Should be `Object.hasOwn` or loop over `Object.entries` instead since the value is used below.